File: 프로그래머스/Level1/콜라츠 추측.cpp
#include <string>
#include <vector>

using namespace std;

int solution(int num) {
    int answer = 0;
    int n = 0;
    int cnt = 0;
    long long a = num;
    while(n < 500){
        
        if(a %2 ==0){
            a = (a/2);
            cnt ++;
            n++;
            if(a==1) break;
            else continue;
        }
        
        else if(a==1)break;
        
        else{
            a = (a*3)+1;
            cnt ++;
            n++;
            if(a==1) break;
            else continue;
        }
    }
    
    if(a == 1){
        answer = cnt;
    }
    else answer = -1;
    
    return answer;
}

File: 프로그래머스/Level1/콜라츠 추측.js
function solution(num) {
    if(num === 1) return 0; 
    
    let k=1;
    while(num !== 1){
        if(k === 500) return -1;
        
        if(num%2 === 0) num /= 2;
        else num = num*3 + 1;
        
        if(num === 1) return k;
        k++;
    }
}
